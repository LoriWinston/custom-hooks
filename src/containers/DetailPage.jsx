import React from 'react';
import { useOneCharacter } from '../hooks/hooks';



const DetailPage = () => {
  const { oneCharacter, loading } = useOneCharacter();

  
  return (
    <div>
      {
        loading
          ?
          <h2>Loading...</h2>
          :
          <div >
            <img src={oneCharacter.photoUrl} alt={name}/>
            <h2>{oneCharacter.name}</h2>
          </div>
      }
    </div>
  );
    
};

export default DetailPage;
