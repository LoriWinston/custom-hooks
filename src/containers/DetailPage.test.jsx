/* eslint-disable max-len */
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import DetailPage from '../containers/DetailPage';
import { MemoryRouter } from 'react-router';
import { Route } from 'react-router-dom';

describe('DetailPage', () => { 
  it('renders a character to the page', async () => { 
    render(
      <MemoryRouter initialEntries={['/details/5cf5679a915ecad153ab68cc']}>
        <Route path="/details/:id"><DetailPage/></Route>
      </MemoryRouter>
    );
        
    const name = await screen.findByText('Afiko');
    expect(name).not.toBeEmptyDOMElement();
    return waitFor(() => {
      expect(name).toMatchSnapshot();
    });
  });
});

