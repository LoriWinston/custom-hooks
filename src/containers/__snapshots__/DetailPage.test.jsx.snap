// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage renders a character to the page 1`] = `
<h2>
  Afiko
</h2>
`;
